charts/ingress-nginx/tests/controller-deployment_test.yaml

```diff
@@ -231,3 +231,46 @@ tests:
       - equal:
           path: spec.template.spec.runtimeClassName
           value: myClass
+
+  - it: should create a Deployment with resizePolicy if `controller.resizePolicy` is set and Kubernetes version >= 1.30
+    capabilities:
+      majorVersion: 1
+      minorVersion: 30
+    set:
+      controller.resizePolicy:
+        - resourceName: cpu
+          restartPolicy: NotRequired
+        - resourceName: memory
+          restartPolicy: RestartContainer
+    asserts:
+      - equal:
+          path: spec.template.spec.containers[0].resizePolicy[0].resourceName
+          value: cpu
+      - equal:
+          path: spec.template.spec.containers[0].resizePolicy[0].restartPolicy
+          value: NotRequired
+      - equal:
+          path: spec.template.spec.containers[0].resizePolicy[1].resourceName
+          value: memory
+      - equal:
+          path: spec.template.spec.containers[0].resizePolicy[1].restartPolicy
+          value: RestartContainer
```

Comment:
`value` can also be a whole array. You do not need to check each element of the array.